src/main/java/com/currencycloud/client/Utils.java

```diff
@@ -1,39 +1,33 @@
 package com.currencycloud.client;
 
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
 public enum Utils {
     ;
 
     public static final String dateFormat = "yyyy-MM-dd'T'HH:mm:ssX";
 
-    public static String join(Iterable<String> strings, String separator) {
-        if (strings == null) {
+    public static String join(List<String> strings, String separator) {
+        if (!validInput(strings, separator)) {
             return null;
         }
-        StringBuilder sb = new StringBuilder();
-        for (String string : strings) {
-            if (sb.length() > 0) {
-                sb.append(separator);
-            }
-            sb.append(string);
-        }
-        return sb.toString();
+        return String.join(separator, strings);
     }
 
     public static String joinInverse(List<String> strings, String separator) {
-        if (strings == null) {
+        if (!validInput(strings, separator)) {
             return null;
         }
-        StringBuilder sb = new StringBuilder();
-        for (int i = strings.size() - 1; i >= 0; i--) {
-            String string = strings.get(i);
-            if (sb.length() > 0) {
-                sb.append(separator);
-            }
-            sb.append(string);
-        }
-        return sb.toString();
+
+        List<String> reversed = new ArrayList<>(strings);
+        Collections.reverse(reversed);
+        return join(reversed, separator);
+    }
+
+    private static boolean validInput(List<String> strings, String separator) {
```

Comment:
would it not be better to have a test "invalidInputs"

`return strings == null || separator == null`

then you would not need to test !validInput(...), but rather invalidInputs(...)